### PR TITLE
DicomAssociation Controller

### PR DESCRIPTION
**Pull request by lillie-dae**
### Description

Implementation for the Dicom Association Controller

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [x] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [x] User guide updated.
- [x] I have updated the changelog

### CODE REVIEW

File: docs/api/rest/dicom-association.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+-->
+
+# DICOM Association information
+
+The `/dai' endpoint is for retrieving a list of information regarding dicom
```
Comment (mocsharp) [inline docs/api/rest/dicom-association.md:19]: How about `/dicom-associations`?

File: src/Configuration/InformaticsGatewayConfiguration.cs [hunk 2]
```diff
@@ -82,6 +82,12 @@ public class InformaticsGatewayConfiguration
         [ConfigurationKeyName("plugins")]
         public PlugInConfiguration PlugInConfigurations { get; set; }
 
+        /// <summary>
+        /// Represents the <c>endpointSettings</c> section of the configuration file.
+        /// </summary>
+        [ConfigurationKeyName("endpointSettings")]
```
Comment (mocsharp) [inline src/Configuration/InformaticsGatewayConfiguration.cs:88]: Can we move this under `InformaticsGatewayConfiguration` > `HttpConfigurations`?
The reason is that we have many other endpoints DIMSE, HL7, etc...
Thanks!

File: src/InformaticsGateway/Logging/Log.0.General.cs [hunk 3]
```diff
@@ -59,5 +59,8 @@ public static partial class Log
 
         [LoggerMessage(EventId = 13, Level = LogLevel.Critical, Message = "Failed to start {ServiceName}.")]
         public static partial void ServiceFailedToStart(this ILogger logger, string serviceName, Exception ex);
+
+        [LoggerMessage(EventId = 14, Level = LogLevel.Error, Message = "Unexpected error occurred in GET /dai API..")]
+        public static partial void DAIControllerGetAllAsyncError(this ILogger logger, Exception ex);
```
Comment (mocsharp) [inline src/InformaticsGateway/Logging/Log.0.General.cs:64]: I think there is a logger file for HTTP API calls?

File: docs/api/rest/dicom-association.md (hunk 1 above)
Comment (lillie-dae) [reply docs/api/rest/dicom-association.md]: Made this change 👍

File: src/Configuration/InformaticsGatewayConfiguration.cs (hunk 2 above)
Comment (lillie-dae) [reply src/Configuration/InformaticsGatewayConfiguration.cs]: Made this change 👍

File: src/InformaticsGateway/Logging/Log.0.General.cs (hunk 3 above)
Comment (lillie-dae) [reply src/InformaticsGateway/Logging/Log.0.General.cs]: Made this change 👍

**Pull request merged**